src/include/ZividPython/DataModelWrapper.h

```diff
@@ -292,6 +290,64 @@ namespace ZividPython
             }
             return pyClass;
         }
+
+        constexpr bool isDirectChild(const std::string_view parent, const std::string_view child)
+        {
+            // Top-level nodes are always direct children of the root
+            if(parent.empty() && child.find('/') == std::string::npos)
+            {
+                return true;
+            }
+            if(child.size() <= parent.size())
+            {
+                return false;
+            }
+            if(child.compare(0, parent.size(), parent) != 0)
+            {
+                return false;
+            }
+            // Ensure there are no additional path components
+            const auto remainingPath = child.substr(parent.size());
+            return remainingPath.find('/') == 0 && remainingPath.find('/', 1) == std::string::npos;
+        }
+
+        static_assert(isDirectChild("", "TopLevel") == true);
+        static_assert(isDirectChild("", "TopLevel/WithChild") == false);
+        static_assert(isDirectChild("TopLevel", "TopLevel/WithChild") == true);
+        static_assert(isDirectChild("TopLevel", "TopLevel/WithChild/GrandChild") == false);
+
+        template<typename DM, typename PyDest, typename Node>
+        void findAndWrapUninstantiatedNodesInDataModel(PyDest &dest, const Node &node)
+        {
+            node.forEach([&](const auto &member) {
+                using MemberType = std::remove_const_t<std::remove_reference_t<decltype(member)>>;
+
+                if constexpr(MemberType::nodeType == Zivid::DataModel::NodeType::group)
+                {
+                    findAndWrapUninstantiatedNodesInDataModel<DM>(dest, member);
+                }
+                else if constexpr(MemberType::nodeType == Zivid::DataModel::NodeType::leafDataModelList)
+                {
+                    using ValueTypeContained = typename MemberType::ValueType::value_type;
+
+                    // Sanity check
+                    static_assert(Zivid::DataModel::IsDataModelType<ValueTypeContained>::value);
+
+                    if constexpr(Zivid::Detail::TypeTraits::IsInTuple<ValueTypeContained,
+                                                                      typename DM::Descendants>::value)
+                    {
+                        // This node is instantiated.
+                        return;
+                    }
+
+                    // Check via path that the contained type is a direct child of the dest
+                    if constexpr(isDirectChild(DM::path, ValueTypeContained::path))
+                    {
+                        wrapDataModel<false>(dest, ValueTypeContained{}, true);
+                    }
```

Comment:
this is okay for now. I wonder if it makes better sense to have `ValueTypeContained` defined as part of `DM::Descendants` for leafDataModelList types. For example, defining `using Settings::Acquisitions::Descendants = std::tuple<Settings::Acquisition>`